### PR TITLE
Hamming weight counter added

### PR DESCRIPTION
**Pull request by ph4r05**
#20 - pls take a look, I've implemented HW counter stream. Pls review if that matches your expectations on the code.

Thanks

### CODE REVIEW

File: streams.h [hunk 1]
```diff
@@ -218,6 +218,73 @@ struct sac_2d_all_pos : stream {
     std::size_t _flip_bit_position;
 };
 
+template<typename value_type>
```
Comment (mitko501) [inline streams.h]: I just realized you shouldn't mix value_type with this: https://github.com/crocs-muni/eacirc-streams/blob/25de00081418995e63fd28a31a9a9c5550040844/stream.h#L9 definition.

Also, why are these function out of hw_counter structure? Why not as a private functions within hw_counter. Do you plan to use those somewhere else?

File: streams.h [hunk 2]
```diff
@@ -218,6 +218,73 @@ struct sac_2d_all_pos : stream {
...
+        } else {
+            std::fill_n(_origin_data.begin(), osize, 0);
+        }
+
+        combination_init(_cur_positions, _cur_hw);
+    }
+
+    vec_cview next() override;
+
+protected:
```
Comment (mitko501) [inline streams.h]: Why not private?

**Comment by Bender250** [comment]
I coded my suggestions in the last commit. Code is working, I tried to debug the edge cases.

If you accept the changes, I almost agree with PR. The only thing I am thinking now is testing - I do not know, how to write tests for this. Code was debugged and it is OK, so maybe we can just note that in the code with comment about @ph4r05 as author and let it be...

**Comment by mitko501** [comment]
Hi @Bender250 I like your changes. But what do you miss in current tests created by @ph4r05? Maybe period test could be split into more tests, but otherwise I find them sufficient.

**Comment by Bender250** [comment]
I am sorry, I completely missed them :(.
